core/worker.go

```diff
@@ -1462,26 +1492,46 @@ func (w *worker) processQiTx(tx *types.Transaction, env *environment) error {
 		return err
 	}
 	// Check tx against required base fee and gas
-	baseFeeInQi := misc.QuaiToQi(env.wo, env.wo.Header().BaseFee())
-	minimumFee := new(big.Int).Mul(baseFeeInQi, big.NewInt(int64(txGas)))
+	minimumFeeInQuai := new(big.Int).Mul(big.NewInt(int64(txGas)), env.wo.BaseFee())
+	minimumFee := misc.QuaiToQi(env.wo, minimumFeeInQuai)
 	if txFeeInQit.Cmp(minimumFee) < 0 {
-		return fmt.Errorf("tx %032x has insufficient fee for base fee of %d and gas of %d", tx.Hash(), baseFeeInQi.Uint64(), txGas)
+		return fmt.Errorf("tx %032x has insufficient fee for base fee, have %d want %d", tx.Hash(), txFeeInQit.Uint64(), minimumFee.Uint64())
 	}
-	log.Global.Infof("Minimum fee: %d", minimumFee.Int64())
 	// Miner gets remainder of fee after base fee
 	txFeeInQit.Sub(txFeeInQit, minimumFee)
 
+	if conversion {
+		// If this transaction has at least one conversion, the tx gas is divided for all ETXs
+		txFeeInQuai := misc.QiToQuai(env.wo, txFeeInQit)
+		gas := new(big.Int).Div(txFeeInQuai, env.wo.BaseFee())
+		gas.Div(gas, big.NewInt(int64(len(etxs))))
+		if gas.Uint64() > (env.wo.GasLimit() / params.MinimumEtxGasDivisor) {
```

Comment:
This calculation will also change once aggregating to a single Quai address.

Reply:
If a user decides to do a conversion and ETX in the same transaction (not recommended) he should supply a lot of gas, so that if gas is distributed equally, the conversion gets enough gas to succeed. There is no way for the user to specify which ETX gets which portion of the gas, so I divide it equally amongst all ETXs.